js/parts/Axis.js

```diff
@@ -1431,9 +1431,7 @@ var Axis = /** @class */ (function () {
         // set the translation factor used in translate function
         axis.setAxisTranslation(true);
         // hook for ordinal axes and radial axes
-        if (axis.beforeSetTickPositions) {
-            axis.beforeSetTickPositions();
-        }
+        fireEvent(this, 'setTickInterval');
         // hook for extensions, used in Highstock ordinal axes
         if (axis.ordinal) {
```

Comment:
Since this block always runs right after `beforeSetTickPositions`, and the ordinal axis already has a hook for that, can we just move this into the same hook?